### PR TITLE
feat(react-dashboard): convert to 0.18

### PR DESCRIPTION
**Pull request by FlowFlorent**
- Ensure presence has correct shape
- `useOthersMapped`
- `useOthersConnectionIds`

### CODE REVIEW

File: examples/react-dashboard/src/liveblocks.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createClient } from "@liveblocks/client";
 import { createRoomContext } from "@liveblocks/react";
 
-let PUBLIC_KEY = "pk_YOUR_PUBLIC_KEY";
+let PUBLIC_KEY = "pk_YOUR_PUBLIC_KEY-X";
```
Comment (FlowFlorent) [inline examples/react-dashboard/src/liveblocks.config.js:4]:
```suggestion
let PUBLIC_KEY = "pk_YOUR_PUBLIC_KEY";
```

File: examples/react-dashboard/src/App.js [hunk 2]
```diff
@@ -44,17 +48,14 @@ function Example() {
...
       return true;
     }
 
-    for (const other of others.toArray()) {
-      if (
-        other.presence?.selectedDataset?.cardId === cardId &&
-        other.presence?.selectedDataset?.dataKey === dataKey
-      ) {
+    for (const other of others) {
+      if (other[1]?.cardId === cardId && other[1]?.dataKey === dataKey) {
```
Comment (nvie) [inline examples/react-dashboard/src/App.js:58]: I personally find the following syntax a little bit more readable:

```suggestion
    for (const [_, selectedDataset] of others) {
      if (selectedDataset?.cardId === cardId && selectedDataset?.dataKey === dataKey) {
```

File: examples/react-dashboard/src/App.js [hunk 3]
```diff
@@ -232,7 +233,14 @@ function Example() {
 
 export default function App() {
   return (
-    <RoomProvider id={roomId} initialPresence={{ cardId: null, cursor: null }}>
+    <RoomProvider
+      id={roomId}
+      initialPresence={{
+        selectedDataset: { cardId: null, dataKey: null },
+        cursor: null,
+        cardId: null,
```
Comment (nvie) [inline examples/react-dashboard/src/App.js:241]: Is this top-level `cardId` field even used anywhere?

**Review by nvie — APPROVED** [review]
Nice work, @FlowFlorent! 💯

File: examples/react-dashboard/src/App.js (hunk 3 above)
Comment (FlowFlorent) [reply examples/react-dashboard/src/App.js:241]: yes it is used here https://github.com/liveblocks/liveblocks/blob/fd6b98461b945646dd3dcceb64d9c0dbcab3e4fb/examples/react-dashboard/src/components/Card.js#L63

File: examples/react-dashboard/src/App.js (hunk 2 above)
Comment (FlowFlorent) [reply examples/react-dashboard/src/App.js:58]: great suggestion, gonna implement

**Pull request merged**